### DISCUSSION TITLE
How do I bundle selected packages under node_modules?

### DISCUSSION THREAD

**Discussion by denik1981** [egoist/tsup#538] [Q&A]
I'm working with a monorepo and the server when it is bundled, especially in watch mode, needs to watch and bundle a few modules that are installed as package dependencies and develop in the same repo. By default, tsup is excluding node_modules from the bundle.

**Comment by transitive-bullshit** [comment]
I also have a use case where this would be ideal.

In particular, I'm publishing an ESM package and certain commonjs dependency imports behave differently in the consuming application depending on a combination of node.js version, webpack version, etc.

I'd love to specify that these commonjs packages should be bundled as opposed to the current functionality of ignoring all `dependencies`.

**Reply by egoist** [reply]
the `noExternal` option might be what you want: `noExternal: [/^package-name($|\/)/]`

**Reply by transitive-bullshit** [reply]
This is exactly what I was looking for. Thanks @egoist 💪
